modin/backends/base/query_compiler.py

```diff
@@ -1035,7 +1035,14 @@ def insert(self, loc, column, value):
         Returns:
             A new QueryCompiler with new data inserted.
         """
-        return DataFrameDefault.register(pandas.DataFrame.insert, inplace=True)(
+
+        def inserter(df, loc, column, value):
+            if isinstance(value, pandas.DataFrame):
```

Comment:
Handling of the frame cases now happens at the front-end so we do not expect to meet DataFrame here unless it was originally Series at the front-end